src/backend/distributed/shared_library_init.c

```diff
@@ -1128,6 +1131,16 @@ RegisterCitusConfigVariables(void)
 		GUC_STANDARD,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.enable_cluster_clock",
+		gettext_noop("When true, distributed transaction is stamped with "
+					 "logical causal clock and returned via the UDF citus_get_transaction_clock()"),
+		NULL,
+		&EnableClusterClock,
+		false,
+		PGC_USERSET,
```

Comment:
Yeah, in the original design, we were stamping all the transactions if this true, since now this in on a need basis, I kept for emergency. If the application is polling for timestamp before every transaction commit, and something is amiss, this will serve as a safety valve. Let me try to change the name/description.

Reply:
Done